### PR TITLE
coursier: Add update script

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Add update script

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/coursier/default.nix [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+  inherit version;
+
   pname = "coursier";
-  version = "2.0.6";
 
   src = fetchurl {
-    url = "https://github.com/coursier/coursier/releases/download/v${version}/coursier";
-    sha256 = "0zg07bi44a5p0zfc77i1vmjz9rwzxzf6p0xsybsv3n1saq884wij";
+    url =
+      "https://github.com/coursier/coursier/releases/download/v${version}/coursier";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/coursier/default.nix:21]:
```suggestion
    url = "https://github.com/coursier/coursier/releases/download/v${version}/coursier";
```

File: pkgs/development/tools/coursier/default.nix [hunk 2]
```diff
@@ -23,12 +30,42 @@ stdenv.mkDerivation rec {
...
+    else
+      echo "${pname} is already up-to-date"
+    fi
   '';
 
   meta = with stdenv.lib; {
     homepage = "https://get-coursier.io/";
-    description = "A Scala library to fetch dependencies from Maven / Ivy repositories";
+    description =
+      "A Scala library to fetch dependencies from Maven / Ivy repositories";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/coursier/default.nix:68]:
```suggestion
    description = "A Scala library to fetch dependencies from Maven / Ivy repositories";
```

File: pkgs/development/tools/coursier/default.nix (hunk 1 above)
Comment (NeQuissimus) [reply pkgs/development/tools/coursier/default.nix:21]: @SuperSandro2000 What is this running? `nixfmt` disagrees with these suggestions.

Comment (SuperSandro2000) [reply pkgs/development/tools/coursier/default.nix:21]: That this line break does not really add anything in keeping the line shorter. Also a line length of 80 is really outdated and only uses 1/5 of most monitors.

**Pull request merged**